quickfixj-core/src/main/java/quickfix/Message.java

```diff
@@ -92,27 +92,26 @@ protected Message(int[] fieldOrder) {
 
     public Message(String string) throws InvalidMessage {
         initializeHeader();
-        fromString(string, null, true, true);
+        fromString(string, null, null, true, true);
     }
 
     public Message(String string, boolean validate) throws InvalidMessage {
         initializeHeader();
-        fromString(string, null, validate, true);
+        fromString(string, null, null, validate, true);
     }
 
-    public Message(String string, DataDictionary dd) throws InvalidMessage {
+    public Message(String string, DataDictionary dd, ValidationSettings dds) throws InvalidMessage {
```

Comment:
Would it be sensible to have a constructor which does `new ValidationSettings()` if the dictionary is not NULL and if the settings are not passed? That way one wouldn't have to change their code if they are using the default validation settings.